api/respond_with_json.go

```diff
@@ -0,0 +1,14 @@
+package main
+
+import "net/http"
+
+const (
+	contentTypeJSON   = "application/json"
+	contentTypeHeader = "Content-Type"
+)
+
+// respondWithJSON sends a response with status code 200.
+func respondWithJSON(w http.ResponseWriter, json []byte) {
```

Comment:
you may want to enforce that you're being sent json or something. You can't really rely on callers to properly marshal their data because some callers might be written by people like me.